src/Microsoft.Health.Fhir.SqlServer/Features/Search/Expressions/Visitors/QueryGenerators/SqlQueryGenerator.cs

```diff
@@ -256,13 +256,28 @@ public object VisitTable(TableExpression tableExpression, SearchOptions context)
                             .Append(" = ").Append(V1.Resource.ResourceId, resourceTableAlias);
                     }
 
+                    // Denormalized predicates on reverse chains need to be applied via a join to the resource table
```

Comment:
Do we need test for paging or is that a separate user story?

Reply:
I added paging tests for the various combinations of chaining/reverse chaining.